storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -526,4 +524,13 @@ private void writeToFile() {
             lock.readLock().unlock();
         }
     }
+
+    private void writeToFileForTruncation() {
```

Comment:
Perhaps it's clearer to rename this and `checkpoint.writeForTruncation` to sth like `writeIfDirExists`.